sympy/sets/ordinals.py

```diff
@@ -0,0 +1,176 @@
+from sympy.core import Basic
+import operator
+
+class Ordinal(Basic):
+    """
+    Represents ordinals in Cantor normal form.
+
+    Internally, this class is just a list of lists (known
+    as "terms").
+    In single terms [a, b], a represents power to the ordinal and b represents multiplying factor
+    Examples
+    ========
+    >>> from sympy.sets import Ordinal, Ordinals
+    >>> Ordinals.w  # doctest: +SKIP
+    w
+    >>> a = Ordinals.w
+    >>> a.is_limit_ordinal
+    True
+    >>> Ordinal([[3,2]])  # doctest: +SKIP
+    {w**3}*2
+    >>> Ordinal([[5,1],[3,2]])  # doctest: +SKIP
+    w**5 + {w**3}*2
+    """
+    def __new__(cls, terms):
+        obj = super(Ordinal, cls).__new__(cls, *terms)
+        powers = [i[0] for i in obj.args]
+        if not all(earlier >= later for earlier, later in zip(powers, powers[1:])):
+            raise ValueError("powers must be in decreasing order")
+        return obj
+
+    @property
+    def is_successor_ordinal(self):
+        return (self.args[-1][0] == 0)
+
+    @property
+    def is_limit_ordinal(self):
+        return not self.is_successor_ordinal
+
+    def _insert_index(self, a, x, lo=0, hi=None):
+        """Return the index where to insert item x in list a,
+        assuming a is sorted in descending order
+
+        Essentially, the function returns number of elements in a which are >= than x.
+        """
+        if lo < 0:
+            raise ValueError('lo must be non-negative')
+        if hi is None:
+            hi = len(a)
+        while lo < hi:
+            mid = (lo+hi)//2
+            if x > a[mid]: hi = mid
+            else: lo = mid+1
+        return lo
+
+    def _compare(self, other, index):
+        value_self = [i[index] for i in self.args]
+        value_other = [i[index] for i in other.args]
+        if value_other == value_self:
+            return -1
+        else:
+            return map(operator.eq, value_self, value_other).index(False)
```

Comment:
Is it enough to only check for equality?

Reply:
Yes , if both the lists are exactly same else its returning the first point of difference.

Reply:
I would have expected that also the multiplicity component should be inspected, not only the equality of the exponent parts. I would have defined a class (maybe `OmegaPower`) whose arguments are `exponent` and `multiplicity` (or, briefly, `exp` and `mult`), and then defined comparisons for them. (The arguments of `Ordinal` would then be objects of that class.)

Reply:
it is both checking multiplicity and power separately. first it is comparing list of exponent parts then list of multiplicity parts. for this a variable `index` is passed to the function` _compare`

Reply:
It seems to me that comparing both at the same time would be less complicated.

Reply:
ok thanks , I will update it .

Reply:
It is also necessary to define something like `OmegaPower(Basic)` since the arguments of a `Basic` object must be `Basic` objects. The arguments are automatically saved by `Basic.__new__`. They can be made more accessible by defining them as properties:

    @property
    def exp(self):
         return self.args[0]